onnxruntime/test/optimizer/qdq_test_utils.h

```diff
@@ -187,5 +187,35 @@ GetQDQTestCaseFn BuildBinaryOpTestCase(const std::vector<int64_t>& input_shape,
                                                 output_arg);
   };
 }
+
+template <typename InputType, typename OutputType>
+GetQDQTestCaseFn BuildQDQTransposeTestCase(
+    const std::vector<int64_t>& input_shape,
+    const std::vector<int64_t>& perms) {
+  return [input_shape, perms](ModelTestBuilder& builder) {
+    auto* input_arg = builder.MakeInput<InputType>(input_shape, -128, 127);
+    auto* output_arg = builder.MakeOutput();
+
+#ifdef USE_NNAPI
+    uint8_t dq_zp = std::numeric_limits<InputType>::max() / 2;
+    uint8_t q_zp = std::numeric_limits<OutputType>::max() / 2;
+#else
+    int8_t dq_zp = 1;
+    int8_t q_zp = 1;
+#endif
+
+    // add DQ
+    auto* dq_output = builder.MakeIntermediate();
+    builder.AddDequantizeLinearNode<InputType>(input_arg, .003f, dq_zp, dq_output);
+
+    // add Transpose
+    auto* transpose_output = builder.MakeOutput();  // transpose output is graph output
+    Node& transpose_node = builder.AddNode("Transpose", {dq_output}, {transpose_output});
```

Comment:
Transpose output will go to Q node, this is not a graph output, this could be the reason UT failes

Reply:
thanks so much for picking this. yes I just realized the reason is that I copied the transpose_nofusion qdqtransformer test cases, looks very similar.  but that one is the one where dq/q should not be dropped... fixed it;)